apps/dashboard/app/models/script.rb

```diff
@@ -159,6 +162,30 @@ def most_recent_job_cluster
 
   private
 
+  def write_job_options_to_cache(opts)
+    File.write(cache_file_path, opts.to_json)
+  end
+
+  def cache_file_path
+    Pathname.new(project_dir).join(".ondemand/scripts/#{id}_opts.json")
+    #Pathname.new("#{Script.scripts_dir(project_dir)}/#{id}_cache.json")
+  end
+
+  def cache_file_exists?
+    cache_file_path.exist?
+  end
+
+  def cached_values
+    @cached_values ||= begin
+      cache_file_path = OodAppkit.dataroot.join(Script.scripts_dir("#{project_dir}"), "#{id}_opts.json")
+      cache_file_content = File.read(cache_file_path) if cache_file_exists?
+      
+      File.exist?(cache_file_path) ? JSON.parse(cache_file_content) : {}
+    rescue => exception
+      raise "Error reading from cache file: #{exception.message}"
```

Comment:
IDK if there's a lot of value in rescuing the error only to throw another. Seems like we should rescue, log a warning, then return an empty hash.